core/cosmosdb/cache-invalidator/src/main/scala/org/apache/openwhisk/core/database/cosmosdb/cache/CacheInvalidator.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.openwhisk.core.database.cosmosdb.cache
+
+import akka.actor.{ActorSystem, CoordinatedShutdown}
+import akka.event.slf4j.SLF4JLogging
+import akka.kafka.ProducerSettings
+import akka.stream.ActorMaterializer
+import com.typesafe.config.Config
+import org.apache.kafka.common.serialization.StringSerializer
+import org.slf4j.bridge.SLF4JBridgeHandler
+
+import scala.concurrent.Future
+import scala.util.Success
+
+object CacheInvalidator extends SLF4JLogging {
+  //CosmosDB changefeed support uses Java Logging.
+  // Those needs to be routed to Slf4j
+  SLF4JBridgeHandler.removeHandlersForRootLogger()
+  SLF4JBridgeHandler.install()
+
+  //TODO Replace with constant from RemoteCacheInvalidation
+  val cacheInvalidationTopic = "cacheInvalidation"
+
+  val instanceId = "cache-invalidator"
+  val whisksCollection = "whisks"
```

Comment:
Can these be taking from configuration?

Reply:
Its fine for them being constant. `instanceId` is used as a marker name in the cache invalidation message sent to Kafka. There is no use of this instanceId apart from fact that controller would compare its own id with this and upon mismatch would invalidate the cache entry

```json
{"instanceId":"cache-invalidator","key":{"mainId":"guest/test-package"}}
```

`whisksCollection` is name of the collection from which change feed is run. This is always `whisks`. User would only be customizing the db name but collection name within db remains same. So fine to be constant